src/services/groups.service.ts

```diff
@@ -0,0 +1,96 @@
+import { PrismaClient, Group } from '@prisma/client';
+import { CreateGroupDto } from '@dtos/group.dto';
+import { HttpException } from '@exceptions/HttpException';
+import { isEmpty } from '@utils/util';
+
+class GroupService {
+  public group = new PrismaClient().group;
+
+  public async findAllGroup(): Promise<Group[]> {
+    const allGroup: Group[] = await this.group.findMany();
+    return allGroup;
+  }
+
+  public async findGroupById(groupId: number): Promise<Group> {
+    if (isEmpty(groupId)) throw new HttpException(400, "groupId is empty");
+
+    const findGroup = await this.group.findUnique({ where: { id: groupId }});
+    if (!findGroup) throw new HttpException(409, "Group doesn't exist");
+
+    return findGroup;
+  }
+
+  public async createGroup(GroupData: CreateGroupDto): Promise<Group> {
+    if (isEmpty(GroupData)) throw new HttpException(400, "GroupData is empty");
+
+    const createGroupData: Group = await this.group.create({ data: { ...GroupData, }});
```

Comment:
Hi there, 

We might want to check if the group already exist before create them? 

So 
```suggestion
  const existingGroup = await this.group.findFirst({ 
    where: { name: GroupData.name } 
  });
  if (existingGroup) throw new HttpException(409, "Group name already exists");
  const createGroupData: Group = await this.group.create({ data: { ...GroupData, }});

```